lang/llvm-devel/Portfile

```diff
@@ -77,66 +83,50 @@ if {${subport} eq "llvm-${llvm_version}"} {
 
     depends_lib         port:libxml2 port:libomp port:llvm-${llvm_version} port:ncurses
     depends_build-append port:swig-python port:doxygen
-
-    # Clang doesn't actually use any of these, but the LLVM makefile
-    # system overlinks its binaries. Remove (or convert to build deps?)
-    # after #46040 is resolved.
-    depends_lib-append  port:libedit port:libffi port:ncurses port:zlib
 }
 
-fetch.type              svn
-svn.revision            365158
-
-depends_extract-append  port:subversion
-
-set compiler_rt_rev     ${svn.revision}
-set libcxx_rev          ${svn.revision}
-set clang-modernize_rev ${svn.revision}
-version                 ${llvm_version}-r${svn.revision}
-worksrcdir              trunk
-svn.url                 https://llvm.org/svn/llvm-project/llvm/trunk
-#worksrcdir              release_${llvm_version_no_dot}
-#svn.url                 https://llvm.org/svn/llvm-project/llvm/branches/release_${llvm_version_no_dot}
+# for devel version only
 default_variants-append +assertions
-#default_variants-append +debug
 
 # the backtik pkgconfig setting in the ffi config line requires pkg-config
 depends_build-append    port:pkgconfig
 
-#version                 ${llvm_version}.0
-#epoch                   1
-#master_sites            https://releases.llvm.org/${version}
-#use_xz                  yes
-#extract.suffix          .tar.xz
-#distfiles               llvm-${version}.src${extract.suffix}
-#worksrcdir              llvm-${version}.src
+worksrcdir              llvm-project/llvm
+patch.dir               ${workpath}/llvm-project/llvm
+
+post-extract {
+    move ${workpath}/llvm-project-${llvm-commit} ${workpath}/llvm-project
 
-if {${distfiles} ne ""} {
     if {${subport} eq "llvm-${llvm_version}"} {
         if {[variant_isset polly]} {
-            distfiles-append     polly-${version}.src${extract.suffix}
+            file rename ${workpath}/llvm-project/polly                        ${worksrcpath}/tools/polly
         }
     } elseif {${subport} eq "clang-${llvm_version}"} {
-        distfiles-append     cfe-${version}.src${extract.suffix} compiler-rt-${version}.src${extract.suffix} libcxx-${version}.src${extract.suffix} clang-tools-extra-${version}.src${extract.suffix}
+        file rename ${workpath}/llvm-project/clang                            ${worksrcpath}/tools/clang
+        file rename ${workpath}/llvm-project/compiler-rt                      ${worksrcpath}/projects/compiler-rt
+        file rename ${workpath}/llvm-project/libcxx                           ${worksrcpath}/projects/libcxx
+    	file rename ${workpath}/llvm-project/libcxxabi                        ${worksrcpath}/projects/libcxxabi
```

Comment:
Its not really a big deal, but just for my info, is this renaming necessary just to get the patches to apply as-is ? I ask just because I didn't need to do this with the git clone approach.

Reply:
Jeremy's portfile, with all the reinplacing, all the patching, and way he enables or does not enable certain features like Polly, etc, is all designed around this kind of source tree layout. 

It's the old way, but LLVM still supports it.

As you know, I want to move to a proper new monorepo build, where projects are individually enabled using "enable_projects", as you saw in my other Portfile, but there is still work to do there to make it all work a way to make it all familiar to everyone, and it was much easier to get this version out the door in a few hours by using Jeremy's existing setup (and probably less startling to do it this way for the higher-ups).

Reply:
For example, llvm will automatically build polly if  the source exists in llvm/tools/polly, but for the monorepo approach, you have to specifically add a configure arg to make llvm go and look for polly in another source folder that is at the same level as llvm. Each project is like that, in one way or another.

Reply:
OK thanks, that makes some sense. Longer term (i.e. by the time 11 comes out say) I think we should simplify all this to use the standard mono build approach. Personally I think for someone looking in from the outside, enabling/disabling features via configuration options is  much more obvious (its how pretty much all builds do it) and if it also removes the need for the mysterious moving of sources around, all the better. But, yes, for now if it helps get something out quicker, this is just fine.

Reply:
I will say that I did not think that with a monorepo build you could still, after the various selected projects were built, go into an individual folder and destroot that single project right from there. Nobody does that, and it is not, really, supposed to work like that. You are supposed to figure out and add the individual targets you want to the install target list and then run that from the monorepo. 

For example, for Apple's clang build, you enable the clang project, but then your build target and destroot target are set (from the monorepo) like this:
```
    build.target            stage2-distribution
    destroot.target         stage2-install-distribution
```

The fact that you can "cheat" like that (destrooting the parts you want in the individual folders) might come in useful, if we insist on having llvm and clang and lldb all be separate projects.

The problem is that because nobody does separate installs of these things, we have to see if these separate targets actually exist.